src/haggadah/events.cljs

```diff
@@ -44,6 +45,15 @@
        (.then (fn [user]  (on-success (.-user user))))
        (.catch (fn [error]  (on-error error) )))))
  
+(re-frame/reg-fx
+ ::query!
+ (fn [{:keys [path on-success on-error order-by]}]
+ (-> (firestore/instance)
```

Comment:
NIT: The event `::fetch-collection!` and `::query!` look very similar.... perhaps there's a way to extract a function?

Reply:
I'll see how I can extract a function from both events.